source/common/router/config_impl.h

```diff
@@ -241,11 +241,17 @@ class VirtualHostImpl : public VirtualHost, Logger::Loggable<Logger::Id::router>
   bool includeAttemptCountInRequest() const override { return include_attempt_count_in_request_; }
   bool includeAttemptCountInResponse() const override { return include_attempt_count_in_response_; }
   const std::vector<ShadowPolicyPtr>& shadowPolicies() const { return shadow_policies_; }
-  const absl::optional<envoy::config::route::v3::RetryPolicy>& retryPolicy() const {
-    return retry_policy_;
+  const OptRef<envoy::config::route::v3::RetryPolicy> retryPolicy() const {
```

Comment:
Should the return type be `OptRef<const T>` (here and for `hedgePolicy()`)?

Reply:
Done